File: bindings/wasm/examples/chain_id.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient, initAsync } from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newTestnet();
const chainId = await client.chainId();
console.log("Chain ID:", chainId);

File: crates/iota-sdk/examples/chain_id.rs
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use iota_sdk::graphql_client::{GraphQLClient, error::GraphQLResult};

#[tokio::main]
async fn main() -> GraphQLResult<()> {
    let client = GraphQLClient::new_testnet();

    let chain_id = client.chain_id().await?;
    println!("Chain ID: {chain_id}");

    Ok(())
}
